src/tutorial/hello-world/README.md

```diff
@@ -20,7 +20,32 @@ Each part will look as follows for the string `"Hello, world!"`:
 0x40            48656c6c6f2c20776f726c642100000000000000000000000000000000000000 // Value "Hello, world!"
 ```
 
-Encoding dynamic values takes alot of work!! In order to return `"Hello, world!"` we must return 96 bytes!.
+Encoding dynamic values takes alot of work!! In order to return `"Hello, world!"` we must return 96 bytes!
+
+You might think that we can populate the memory in sequential order starting with the location (0x00), then length (0x20), and then the value (0x40), however, this would result in the value "Hello, world!" being left padded, so we would get: 
+```
+[Byte number]   [DATA]    
+0x00            0000000000000000000000000000000000000000000000000000000000000020 // The location of the "Hello, world!" data (dynamic type).
+0x20            000000000000000000000000000000000000000000000000000000000000000d // The length of "Hello, world!" in bytes
+0x40            0000000000000000000000000000000000000048656c6c6f2c20776f726c6421 // Wrong encoding!
+```
+And that's not what we need. Instead, we're going to leverage what we know about memory to produce the right result, and that is:
+- Memory is always expanded in 32 byte increments
+- We can store a value starting from any index
+- We want to store the location starting at 0x00 and the length starting at 0x20
+- Values are left padded
+
+Armed with that knowledge, we can store the value "Hello, world!" at 0x2d (starting index of length 0x20 + the length of the data in bytes 0x0d). That will give us the first 0x2d (decimal 45) bytes as zeroes due to memory expansion, and since the value is left padded and of length 0x0d (decimal 13) bytes we get another 19 bytes of zeroes, equalling to 64 bytes. 
+
+This means our actual important bytes start at 0x40, but since memory is expanded in 32 byte increments we get right padded zeroes until 0x60, equalling 96 bytes, exactly what we need! 
```

Comment:
```suggestion
This means our important bytes (`48656c6c6f2c20776f726c6421") start at 0x40. However, since memory is expanded in 32 byte increments empty memory will be expanded with zeros to byte 0x60 (decimal 96), exactly what we need! 
```